### DISCUSSION TITLE
Unable to connect to minio

### DISCUSSION THREAD

**Discussion by habibutsu** [apache/airflow#33533] [General]
### Apache Airflow version

2.6.3

### What happened

[2023-07-26T21:28:19.052+0000] {base_aws.py:160} INFO - No connection ID provided. Fallback on boto3 credential strategy (region_name='\x1b[01mNone\x1b[22m'). See: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html
Out[19]: 
(False,
 "'NoCredentialsError' error occurred while testing connection: Unable to locate credentials")

### What you think should happen instead

Connection works

### How to reproduce

```python
from airflow.models.connection import Connection
conn = Connection.from_json('{"conn_type": "aws", "login": "minio-user", "password": "minio-password", "extra":{"endpoint_url": "http://minio-storage:9000"}}')
conn.test_connection()
````

### Operating System

Linux/docker

### Versions of Apache Airflow Providers

apache-airflow==2.6.3
apache-airflow-providers-amazon==8.2.0

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by habibutsu** [comment]
As I see [here](https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/hooks/base_aws.py#L813) is executing following code:
```python
 conn_info = session.client(
    "sts",
    endpoint_url=test_endpoint_url,
).get_caller_identity()
```

And of course it does not work for minio

**Comment by Taragolis** [comment]
This case scenario described in [Amazon Web Services Connection](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html#amazon-web-services-connection) documentation

![image](https://github.com/apache/airflow/assets/3998685/4d451343-c200-4c80-8c1a-a1e5589bd9e3)

**Reply by hanleybrand** [reply]
This is definitely true for min.io -- the test has always failed for me. I don't know if its possible to configure min.io in a way that it would pass the test, but by default it won't pass. I haven't successfully upgraded to 2.6.x yet, but with 2.5.2 I have min.io setup both for data storage and remote logging and it works fine.

**Reply by Taragolis** [reply]
I don't think so. MinIO doesn't implement AWS STS GetCallerIdentity in their [AWS STS Endpoint](https://min.io/docs/minio/linux/developers/security-token-service.html) so it is not possible to utilise test connection by same way it is used in AWS Provider.

**Comment by potiuk** [comment]
Converting to a discussion, as it seems more of a discussion on how to do specific configuration, not an Airlfow Issue per-se.

**Comment by hanleybrand** [comment]
Once you have the connection setup, and assuming you have your min.io key/secret in env vars, try this code to test the conn:

```python
minio_key = os.environ["MINIO_USER"]
minio_secret = os.environ["MINIO_KEY"]
minio_region = os.environ["MINIO_REGION"]
minio_endpoint = os.environ["MINIO_ENDPOINT_URL"]

client = boto3.client(
    's3',
    aws_access_key_id=minio_key,
    aws_secret_access_key=minio_secret,
    region_name=minio_region,
    endpoint_url=minio_endpoint,
)
# check for buckets (assuming you have buckets)
bucket_list = []
lb = client.list_buckets()
for b in lb['Buckets']:
    bucket_list.append(b['Name'])
print(bucket_list)

# upload a file to the first bucket in bucket list - I'm assuming this file exists for you, if not pick another file
file_name = 'dag_processor_manager.log'
file = f"/opt/airflow/logs/dag_processor_manager/{file_name}"

# upload the file
client.upload_file(file, bucket_list[0], file_name)
```

**Reply by hanleybrand** [reply]
BTW - I'm not sure if it makes a difference, but everything I've seen indicated that putting the key/secret in extra was the way to go, and this is how I have it configured

```python
import json

extra = {
    "aws_access_key_id": minio_key,
    "aws_secret_access_key": minio_secret,
    "region_name": minio_region,
    "endpoint_url": minio_endpoint
}

# copy/paste the json.dumps(extra )result into the extra field 
# and leave AWS Access Key ID / AWS Secret Access Key blank

json.dumps(extra)

```
